agglayer/grpc/agglayer_grpc_client.go

```diff
@@ -46,14 +55,29 @@ func NewAgglayerGRPCClient(serverAddr string) (*AgglayerGRPCClient, error) {
 
 // GetEpochConfiguration returns the epoch configuration from the AggLayer
 func (a *AgglayerGRPCClient) GetEpochConfiguration(ctx context.Context) (*types.ClockConfiguration, error) {
-	response, err := a.cfgService.GetEpochConfiguration(ctx, &v1.GetEpochConfigurationRequest{})
+	var (
+		response *v1.GetEpochConfigurationResponse
+		err      error
+	)
+
+	err = aggkitcommon.RetryWithExponentialBackoff(ctx, maxRequestRetries, initialDelay, func() error {
+		response, err = a.cfgService.GetEpochConfiguration(ctx, &v1.GetEpochConfigurationRequest{})
+		if err != nil {
```

Comment:
can we somehow de deuplicate this with other places?

Reply:
Let me know if can be done better: https://github.com/agglayer/aggkit/pull/524/commits/9b5302c45b4293afabae09805691d4909200f02d